client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/MultiServerStoreQueryTest.java

```diff
@@ -0,0 +1,170 @@
+package org.infinispan.client.hotrod.query;
+
+import static org.infinispan.server.hotrod.test.HotRodTestingUtil.hotRodCacheConfiguration;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
+import org.infinispan.client.hotrod.test.MultiHotRodServersTest;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.cache.Index;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.persistence.dummy.DummyInMemoryStoreConfigurationBuilder;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.annotations.ProtoDoc;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoSchemaBuilder;
+import org.infinispan.query.indexmanager.InfinispanIndexManager;
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+import org.testng.annotations.Test;
+
+@Test(testName = "client.hotrod.query.MultiServerStoreQueryTest", groups = "functional")
```

Comment:
I'll add a note here saying this tests for issue ISPN-9068. I'm sure we'll stumble upon more QueryInterceptor oddities in the future again :)

Reply:
I usually don't put JIRAs in the code because that information is already present in the commit.

Reply:
Me neither. But this is an exotic specimen :)